modules/imports.py

```diff
@@ -797,10 +798,13 @@ def parse_folder(self, curr_folder):
                 self.node_add(wiki_string.decode(cons.STR_UTF8), node_name, curr_folder)
                 # check if the node has children
                 children_folder = os.path.join(curr_folder, node_name)
-                if os.path.isdir(children_folder):
-                    self.parse_folder(children_folder)
```

Comment:
This code allows to recognise that if "nodename.txt" exists and also at the same level exists a folder "nodename", the .txt files inside the folder "nodename" will be direct children of the node resulting from "nodename.txt". With your code we lose this but we get that every foldername is parsed also if there is not a foldername.txt beside that. I understand your goal but I would rather add the second part without removing the first, just being careful to do not parse the same folder twice, so just excluding from the second parse a folder that also has a foldername.txt at the same level